interface/patient_file/summary/disclosure_full.php

```diff
@@ -85,7 +85,7 @@ class="css_button" onclick="top.restoreSession()"> <span><?php echo xlt('View Pa
 <br>
 <?php
 $N=15;
-$offset = $_REQUEST['offset'];
+$offset = intval($_REQUEST['offset']);
```

Comment:
best to secure this in script below (there's 1 sql query and i think 2 places where need to html escape)

Reply:
for the sql query, use the escape_limit() function:
https://github.com/openemr/openemr/blob/master/library/formdata.inc.php#L49

Reply:
so line will look like this:
``````
" ORDER BY date DESC LIMIT " . escape_limit($offset) . ", " . escape_limit($N);

Reply:
and for the html escaping, in lines 162 and 172, replace `($offset-$N)` to following:
``````
attr($offset-$N)

Reply:
Also, to clean up this script, remove the commented out query in line 100

Reply:
offset is directly used in this script at line 105 in disclosure_full.php
should we apply the fix here ? 
![image](https://user-images.githubusercontent.com/20621444/33803012-6cf70ac8-dd8e-11e7-9807-59ddff333769.png)

Reply:
yep, see my note above, which states to make it:
``````
" ORDER BY date DESC LIMIT " . escape_limit($offset) . ", " . escape_limit($N);

Reply:
Thanks , I applied and committed it , Hopefully one more down :)

Reply:
I would stay away from using intval above since you just never know when a developer needs to do strange things (for example, in some places developers use 'all' for limit which the code then uses to flag as not using a limit in the query). Just need to ensure the downstream elements are appropriately dealt with (which they are now).

Reply:
Ok I removed it